### PR TITLE
Validate IWA options

### PR DESCRIPTION
**Pull request by sonkkeli**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -72,12 +72,35 @@ function maybeSignWebBundle(webBundle, opts) {
...
+        `Base URL (${opts.baseURL}) doesn't match with the provided private key. Base URL should start with ${expectedOrigin}`
+      );
+    }
+  }
+}
 
+function validateOptions(opts) {
   if (opts.baseURL !== '' && !opts.baseURL.endsWith('/')) {
-    throw new Error("Non-empty baseURL must end with '/'.");
+    throw new Error('Non-empty base URL must end with "/".');
```
Comment (cmfcmf) [inline lib/index.js:95]: IMHO you should keep the option's name as is, i.e.,

> 'Non-empty "baseURL" option must end with "/"'

File: lib/index.js [hunk 2]
```diff
@@ -72,12 +72,35 @@ function maybeSignWebBundle(webBundle, opts) {
...
   }
+}
+
+export default function wbnOutputPlugin(opts) {
+  opts = Object.assign({}, defaults, { baseURL: '' }, opts);
```
Comment (cmfcmf) [inline lib/index.js:100]: You didn't change this, but I see no reason why `baseURL` is not part of the `defaults` object.

File: lib/index.js [hunk 3]
```diff
@@ -72,12 +72,35 @@ function maybeSignWebBundle(webBundle, opts) {
...
+  validateOptions(opts);
+  if (opts.integrityBlockSign) validateIWAOptions(opts);
```
Comment (cmfcmf) [inline lib/index.js:102]: Super small nitpick, but I think it would be a bit nicer to move this `if` condition inside `validateOptions` so that we have a single function you can call to validate all of your options.

File: lib/index.js [hunk 4]
```diff
@@ -72,12 +72,35 @@ function maybeSignWebBundle(webBundle, opts) {
...
+  }
+
+  if (opts.baseURL !== '') {
+    const expectedOrigin = new WebBundleId(
+      parsePemKey(opts.integrityBlockSign.key)
+    ).serializeWithIsolatedWebAppOrigin();
+
+    if (!opts.baseURL.startsWith(expectedOrigin)) {
+      throw new Error(
+        `Base URL (${opts.baseURL}) doesn't match with the provided private key. Base URL should start with ${expectedOrigin}`
```
Comment (cmfcmf) [inline lib/index.js:87]: Maybe reword this to:

> The base URL specified in the "baseURL" option must begin with the base URL derived from the provided private key (${expectedOrigin}), but was "${opts.baseURL}".

Additionally, I am not sure whether we should allow specifying a baseURL that just starts with, but does not match the expected origin. Especially given that this would mean that the manifest is not at `/manifest.webmanifest`, but at some sub path `/foo/manifest.webmanifest`, which doesn't work for us. If you agree, then I'd change the `startsWith` to a `===` check and reword the message as:

> The provided "baseURL" option (${opts.baseURL}) does not match the expected base URL (${expectedOrigin}), which is derived from the provided private key.

File: lib/index.js (hunk 2 above)
Comment (sonkkeli) [reply lib/index.js:102]: I will create a separate PR of this one and add Kunihiko&Reilly to review to be sure :)

**Comment by sonkkeli** [comment]
PTAL

**Review by cmfcmf — APPROVED** [review]
Thanks, looks good 👍
